sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSql.java

```diff
@@ -47,22 +36,23 @@
 Pipeline p = Pipeline.create(options);
 
 //create table from TextIO;
-PCollection<BeamSqlRow> inputTableA = p.apply(TextIO.read().from("/my/input/patha"))
-    .apply(...);
-PCollection<BeamSqlRow> inputTableB = p.apply(TextIO.read().from("/my/input/pathb"))
-    .apply(...);
+PCollection<Row> inputTableA = p.apply(TextIO.read().from("/my/input/patha")).apply(...);
+PCollection<Row> inputTableB = p.apply(TextIO.read().from("/my/input/pathb")).apply(...);
 
 //run a simple query, and register the output as a table in BeamSql;
 String sql1 = "select MY_FUNC(c1), c2 from PCOLLECTION";
-PCollection<BeamSqlRow> outputTableA = inputTableA.apply(
-    BeamSql.query(sql1)
-    .withUdf("MY_FUNC", MY_FUNC.class, "FUNC"));
+PCollection<Row> outputTableA = inputTableA.apply(
+    BeamSql
+        .query(sql1)
+        .registerUdf("MY_FUNC", MY_FUNC.class, "FUNC")
+        .toPTransform());
 
 //run a JOIN with one table from TextIO, and one table from another query
-PCollection<BeamSqlRow> outputTableB = PCollectionTuple.of(
-    new TupleTag<BeamSqlRow>("TABLE_O_A"), outputTableA)
-                .and(new TupleTag<BeamSqlRow>("TABLE_B"), inputTableB)
-    .apply(BeamSql.queryMulti("select * from TABLE_O_A JOIN TABLE_B where ..."));
+PCollection<Row> outputTableB =
+    PCollectionTuple
+        .of(new TupleTag<>("TABLE_O_A"), outputTableA)
```

Comment:
<!--thread_id:cc_167124908_t; commit:fb7b4f8a57280c437bf81303df183270ee2b810d; resolved:0-->
<!--section:context-quote-->
> **kennknowles** wrote:
> The point of `PCollectionTuple` is to allow a heterogeneous map where when you pull the `PCollection` back out, you have strong typing guarantees. (TBH I don't think that is so important anymore as pipeline validation time is when a lot of errors get caught)
> 
> The other benefit that `PCollectionTuple` can offer is that the tuples can have captured type information by using the `new TupleTag<XYZ>() {}` trick. We use this to infer coders, for example. (If Java reflection "just worked" then we would not need this). It isn't relevant for SQL on `PCollection<Row>`. But this connects with a conversation we had the other day: if you have `PCollection<MyPojo>` where `MyPojo` has a registered schema, then the SQL transform could accept it directly and insert necessary conversions to support table-oriented operations.

<!--section:body-->
Kept the us of `PCollectionTuple` unchanged for now.